src/crawlers/cheerio_crawler.js

```diff
@@ -420,6 +420,7 @@ class CheerioCrawler {
 
                 return false;
             },
+            timeoutSecs: this.requestTimeoutMillis / 1000 + 1,
```

Comment:
why `+ 1` ??? This way the timeout will always be at least one second, which doesn't seem right

Reply:
I think that we can just pass the timeout down to `requestAsBrowser` and let it throw and handle that Error, instead of having the `addTimeoutToPromise` call in `_handleRequestFunction`.

Reply:
We should probably throw a `TimeoutError` from `requestAsBrowser` and then catch it and rethrow a new Error from CheerioCrawler, so that the log does not confuse ppl with a timeout error from an unrelated function.